### PR TITLE
Introduce HyperShift-Kubevirt conformance informing job

### PR DESCRIPTION
**Pull request by orenc1**
This periodic job will run the minimal OCP conformance suite to ensure basic functionality on a kubevirt-provided HostedCluster, as well as kubevirt-specific test cases, labeled as `[sig-kubevirt]`.

### CODE REVIEW

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-release-master-hypershift-4.18-periodic-ci-openshift-hypershift-kubevirt-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
1/1 passed.
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
2/2 passed.
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml [hunk 1]
```diff
@@ -153,8 +154,12 @@ chain:
       if [[ -n "${TEST_SKIPS}" ]]; then
           TESTS="$(openshift-tests run --dry-run --provider "${TEST_PROVIDER}" "${TEST_SUITE}")"
           echo "${TESTS}" | grep -v "${TEST_SKIPS}" >/tmp/tests
+          if [[ "${CLUSTER_TYPE}" == "kubevirt" ]]; then
+            echo "${TESTS}" | grep "\[sig-kubevirt\]" >>/tmp/tests
+            KUBVIRT_NO_SKIP="| grep -v '\[sig-kubevirt\]'"
+          fi
           echo "Skipping tests:"
-          echo "${TESTS}" | grep "${TEST_SKIPS}" || { exit_code=$?; echo 'Error: no tests were found matching the TEST_SKIPS regex:'; echo "$TEST_SKIPS"; return $exit_code; }
+          echo "${TESTS}" | grep "${TEST_SKIPS}" ${KUBEVIRT_NO_SKIP:-} || { exit_code=$?; echo 'Error: no tests were found matching the TEST_SKIPS regex:'; echo "$TEST_SKIPS"; return $exit_code; }
```
Comment (davidvossel) [inline ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml:162]: i'm unsure about this approach. It appears this short circuits the ability to skip a `sig-kubevirt` test using the `TEST_SKIPS` env var.

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

File: ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml (hunk 1 above)
Comment (orenc1) [reply ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml]: good catch, thanks.
I've modified the script to take that into account as well, so `TEST_SKIPS` will be honored if running on kubevirt platform.
in a nutshell, the logic is as follows:

1. get the skipped tests list excluding the trailing `[openshift/conformance/parallel]` (if exists), and store in an env var.
2. strip the trailing `\|` from that env var.
3. remove backslashed from that env var.
4. from ALL tests, extract the `[sig-kubevirt]` ones, and filter out the kubevirt skipped tests, using perl regex with grep). non-kubevirt skips were handled earlier.
5. if no `TEST_SKIPS` has been provided, we filter out nothing and add all `[sig-kubevirt]` test cases to the minimal suite.

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Comment by orenc1** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.18-periodics-hypershift-kubevirt-aws-conformance

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

**Comment by orenc1** [comment]
Thanks for the review, David.
I'll monitor the performance of this job daily.
/pj-rehearse ack

**Pull request merged**